[PE-D][Tester B] [RuntimeException] Unhandled DuplicateVehicleException from addvehicle command

Command:
```
addvehicle p/SBA1234A b/Toyota c/1 cl/red t/car 
addvehicle p/SBA1234A b/Toyota c/1 cl/any t/car 
```
Stacktrace:
```
Caused by: seedu.address.model.service.exception.DuplicateVehicleException: Operation would result in duplicate vehicles
        at seedu.address.model.service.UniqueVehicleList.add(UniqueVehicleList.java:49)
        at seedu.address.model.entity.shop.Shop.addVehicle(Shop.java:433)
        at seedu.address.model.ModelManager.addVehicle(ModelManager.java:311)
        at seedu.address.logic.commands.AddVehicleCommand.executeUndoableCommand(AddVehicleCommand.java:67)
        at seedu.address.logic.commands.RedoableCommand.execute(RedoableCommand.java:60)
        at seedu.address.logic.LogicManager.execute(LogicManager.java:61)
        at seedu.address.ui.MainWindow.executeCommand(MainWindow.java:327)
        at seedu.address.ui.CommandBox.handleCommandEntered(CommandBox.java:45)
```

No error feedback from UI.


<!--session: 1680242570588-dcb219bd-e8e2-4524-ba78-c2ef5445b59f--><!--Version: Web v3.4.7-->

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: c0j0s/ped#5

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -80,11 +80,13 @@ Format: `addservicetech s/SERVICE_ID t/TECHNICIAN_ID`
 Example: `addservicetech s/1 t/3`
 
 ### Adding a vehicle: `addvehicle`
-Adds a vehicle of specified type (i.e. motorbike, car) to the system and assigns a unique vehicle ID.
+Adds a vehicle of specified type (i.e. motorbike, car) to the system and assigns a unique vehicle ID. 
 
-Format: `addvehicle p/PLATE_NUMBER b/BRAND c/CUSTOMER_ID t/TYPE`
+Format: `addvehicle p/PLATE_NUMBER b/BRAND c/CUSTOMER_ID cl/COLOR t/TYPE`
 
-Example: `addvehicle p/SBA1234A b/Toyota c/1 t/car`
+Examples: 
+* `addvehicle p/SBA1234A b/Toyota c/1 cl/red t/car`
+* `addvehicle p/SBG5678B b/Honda c/2 cl/blue t/motorbike`
 
 ### Adding a vehicle part: `addpart`
 Adds a vehicle part to the system.
@@ -312,9 +314,6 @@ to be updated
 --------------------------------------------------------------------------------------------------------------------
 
 ## Command summary
-<div markdown="span" class="alert alert-warning">:exclamation: **Caution:**
-Command summary may be using outdated syntax please refer to guide above
-</div>
 
 | Action                                                  | Format, Examples                                                                                                                                                                |
 |---------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

--- a/docs/UserGuide.md
+++ b/docs/UserGuide.md
@@ -53,6 +53,14 @@ AutoM8 is a **desktop app for an auto repair shop, optimized for use via a Comma
 
 </div>
 
+<div markdown="block" class="alert alert-info">
+
+**:information_source: Notes command GUI behaviors:**<br>
+
+* Using entity specific commands (e.g. commandX, where command is a specific operation such as add, edit, etc and X is a specific entity such as customer, vehicle etc) will help you navigate to the respective tabs (e.g. A successful `addvehicle` command will navigate you to the Vehicles tab)
+
+</div>
+
 ---
 
 ### Add
@@ -132,10 +140,15 @@ Example: `addappointmenttech a/1 t/2`
 ---
 ### List/Sort
 
-### Listing all vehicles/customers/parts/appointments: `list`
-Shows all vehicles/customers/parts/appointments.
+### Listing all vehicles/customers/parts/appointments/services/technicians: `list`
+Shows all vehicles/customers/parts/appointments/services/technicians and helps navigate you to the respective tab.
 
-Format: `list(vehicles/customers/parts/appointments/services/technicians)`
+We also have a global `list` command. The "list" command helps to display all entities in their respective tabs while staying on the current tab. Just keep in mind that it doesn't switch tabs for you, so you'll need to do that manually by clicking or using the specific `listX` commands available to help you navigate to the respective tabs.
+
+Format: 
+* `list(vehicles/customers/parts/appointments/services/technicians)` 
+* `list`
+  
 
 Examples:
 * `listvehicles`

--- a/src/main/java/seedu/address/logic/commands/AddVehicleCommand.java
+++ b/src/main/java/seedu/address/logic/commands/AddVehicleCommand.java
@@ -13,7 +13,7 @@ import seedu.address.model.Model;
 import seedu.address.model.service.Vehicle;
 
 /**
- * Manages adding vehicles
+ * Adds a vehicle to the shop
  */
 public class AddVehicleCommand extends RedoableCommand {
     public static final String COMMAND_WORD = "addvehicle";
@@ -32,15 +32,13 @@ public class AddVehicleCommand extends RedoableCommand {
             + PREFIX_VEHICLE_TYPE + "4wd ";
 
     public static final String MESSAGE_SUCCESS = "New vehicle added: %1$s";
-    public static final String MESSAGE_DUPLICATE_VEHICLE = "This vechicle already registered";
+    public static final String MESSAGE_DUPLICATE_VEHICLE = "This vehicle is already registered";
     public static final String MESSAGE_CUSTOMER_NOT_FOUND = "Customer not registered";
 
     private final Vehicle toAdd;
 
     /**
-     * Constructs command that adds vehicle to the model
-     *
-     * @param vehicle Customer to be added
+     * Creates an AddCommand to add the specified {@code Vehicle}
      */
     public AddVehicleCommand(Vehicle vehicle) {
         this.toAdd = vehicle;
@@ -60,7 +58,7 @@ public class AddVehicleCommand extends RedoableCommand {
             IdGenerator.setVehicleIdUnused(toAdd.getId());
             throw new CommandException(MESSAGE_CUSTOMER_NOT_FOUND);
         }
-        if (model.hasVehicle(toAdd.getId())) {
+        if (model.hasVehicle(toAdd)) {
             throw new CommandException(MESSAGE_DUPLICATE_VEHICLE);
         }
 

--- a/src/main/java/seedu/address/model/Model.java
+++ b/src/main/java/seedu/address/model/Model.java
@@ -181,11 +181,18 @@ public interface Model {
     void addVehicle(int customerId, Vehicle vehicle);
 
     /**
-     * Checks if shop already has vehicle
+     * Checks if shop already has vehicle using id
      *
      * @param vehicleId Vehicle ID to check against
      */
     boolean hasVehicle(int vehicleId);
+
+    /**
+     * Checks if shop already has vehicle using plate numbers
+     *
+     * @param vehicle Vehicle to check against
+     */
+    boolean hasVehicle(Vehicle vehicle);
 
     void deleteVehicle(Vehicle target);
 

--- a/src/main/java/seedu/address/model/ModelManager.java
+++ b/src/main/java/seedu/address/model/ModelManager.java
@@ -314,14 +314,14 @@ public class ModelManager implements Model {
         updateFilteredCustomerList(PREDICATE_SHOW_ALL_CUSTOMERS);
     }
 
-    /**
-     * Checks if shop already has vehicle
-     *
-     * @param vehicleId Vehicle ID to check against
-     */
     @Override
     public boolean hasVehicle(int vehicleId) {
         return this.shop.hasVehicle(vehicleId);
+    }
+
+    @Override
+    public boolean hasVehicle(Vehicle vehicle) {
+        return this.shop.hasVehicle(vehicle);
     }
 
     @Override

--- a/src/test/java/seedu/address/logic/commands/AddCommandTest.java
+++ b/src/test/java/seedu/address/logic/commands/AddCommandTest.java
@@ -218,6 +218,11 @@ public class AddCommandTest {
         }
 
         @Override
+        public boolean hasVehicle(Vehicle vehicle) {
+            throw new AssertionError("This method should not be called.");
+        }
+
+        @Override
         public void deleteVehicle(Vehicle target) {
             throw new AssertionError("This method should not be called.");
         }

